samples: Add support for nRF54lm20a to Matter and OpenThread (#22207)
nrfconnect/sdk-nrf: samples/matter/template/README.rst

The development kits for this sample offer the following IPv6 network support for Matter:

> * Matter over Thread is supported for the ``nrf52840dk/nrf52840``, ``nrf5340dk/nrf5340/cpuapp``, ``nrf21540dk/nrf52840``, ``nrf54l15dk/nrf54l15/cpuapp``, ``nrf54l15dk/nrf54l10/cpuapp`` and ``nrf54lm20pdk/nrf54lm20a/cpuapp`` board targets.

Reviewer: ```suggestion
* Matter over Thread is supported for the ``nrf52840dk/nrf52840``, ``nrf5340dk/nrf5340/cpuapp``, ``nrf21540dk/nrf52840``, ``nrf54l15dk/nrf54l15/cpuapp``, ``nrf54l15dk/nrf54l10/cpuapp``, and ``nrf54lm20pdk/nrf54lm20a/cpuapp`` board targets.
```

Reviewer 2: Certainly, thanks!

> * Matter over Thread is supported for the ``nrf52840dk/nrf52840``, ``nrf5340dk/nrf5340/cpuapp``, ``nrf21540dk/nrf52840``, ``nrf54l15dk/nrf54l15/cpuapp``, ``nrf54l15dk/nrf54l10/cpuapp``, and ``nrf54lm20pdk/nrf54lm20a/cpuapp`` board targets.

Reviewer: ```suggestion
* Matter over Thread is supported for the following board targets:

  * ``nrf52840dk/nrf52840``
  * ``nrf5340dk/nrf5340/cpuapp``
  * ``nrf21540dk/nrf52840``
  * ``nrf54l15dk/nrf54l15/cpuapp``
  * ``nrf54l15dk/nrf54l10/cpuapp``
  * ``nrf54lm20pdk/nrf54lm20a/cpuapp``
```
This might improve the readability a bit.

Matter template using only internal memory
==========================================

> For the nRF54L15 and nRF54L20 DKs, you can configure the sample to use only the internal RRAM for storage.

Reviewer: ```suggestion
For the nRF54L15 DK, you can configure the sample to use only the internal RRAM for storage.
```
